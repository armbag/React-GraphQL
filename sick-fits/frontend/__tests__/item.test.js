import ItemComponent from '../components/Item'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

const fakeItem = {
	id: 'ABC123',
	title: 'cool item',
	price: 4000,
	description: 'This item is really cool!',
	image: 'dog.jpg',
	largeImage: 'largedog.jpg',
}

describe('<Item />', () => {
	it('renders and matchs the snapshot', () => {
		const wrapper = shallow(<ItemComponent item={fakeItem} />)
		expect(toJSON(wrapper)).toMatchSnapshot()
	})
	// it('renders the image properly', () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />)
	// 	const img = wrapper.find('img')
	// 	expect(img.props().src).toBe(fakeItem.image)
	// 	expect(img.props().alt).toBe(fakeItem.title)
	// })

	// it('renders the priceTag and title', () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />)
	// 	const PriceTag = wrapper.find('PriceTag')
	// 	// console.log(PriceTag.dive().text())
	// 	// console.log(PriceTag.children().text())
	// 	expect(PriceTag.children().text()).toBe('$5')
	// 	expect(wrapper.find('Title a').text()).toBe(fakeItem.title)
	// })

	// it('renders out the buttons properly', () => {
	// 	const wrapper = shallow(<ItemComponent item={fakeItem} />)
	// 	const buttonList = wrapper.find('.buttonList')
	// 	expect(buttonList.children()).toHaveLength(3)
	// 	expect(buttonList.find('DeleteItem')).toBeTruthy()
	// })
})
