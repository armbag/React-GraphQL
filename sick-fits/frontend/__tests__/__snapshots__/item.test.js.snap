// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> renders and matchs the snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="cool item"
    src="dog.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        cool item
      </a>
    </Link>
  </Title>
  <PriceTag>
    $40
  </PriceTag>
  <p>
    This item is really cool!
  </p>
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": "ABC123",
          },
        }
      }
    >
      <a>
        Edit âœŽ
      </a>
    </Link>
    <AddToCart
      id="ABC123"
    />
    <DeleteItem
      id="ABC123"
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
